### PR TITLE
Q029 with MSA name

### LINKED ISSUE #211: Validation (2017): Q029 (MSA/MD,State,County,Census Tract)
**Edit #:** Q029
**Type:** Quality
**Category:** Loan Application Register (only)
**Column:** MSA/MD,State,County,Census Tract

If the reported state/county combination (when county is small and reported tract equals NA), or state/county/census tract combination is valid then the MSA/MD should not = NA.

Additional resources:
- [FFIEC's 2016 HMDA File Spec](https://www.ffiec.gov/hmda/pdf/spec2016.pdf)
- [FFIEC's 2016 HMDA Edits](https://www.ffiec.gov/hmda/pdf/edit2016.pdf)
- [CFPB's 2017 HMDA File Spec](http://www.consumerfinance.gov/hmda/static/for-filers/2017/2017-HMDA-File-Specifications.pdf)

### PR DESCRIPTION
**Pull request by nickgrippin**
This PR also has some small fixes for the `edit-metadata.txt` file, along with removing a redundant dependency for the API.

Closes #211

### CODE REVIEW

**Comment by schbetsy** [comment]
Looks like the tests can't compile. 

(FYI, in case it helps: you can use `test:compile` in `sbt` to pre-compile the specs)

File: api/src/main/scala/hmda/api/http/ValidationErrorConverter.scala [hunk 1]
```diff
@@ -49,7 +50,11 @@ trait ValidationErrorConverter {
 
     val jsVals: Seq[(String, JsValue)] = fieldNames.map { fieldName =>
       val row = relevantRow(err, vs)
-      val fieldValue = row.valueOf(fieldName)
+      val fieldValue = if (fieldName == "Metropolitan Statistical Area / Metropolitan Division Name") {
+        CbsaLookup.nameFor(row.valueOf("Metropolitan Statistical Area / Metropolitan Division").toString)
```
Comment (schbetsy) [inline api/src/main/scala/hmda/api/http/ValidationErrorConverter.scala]: This needs a spec to show that it produces the correct JSON response for a Q029 failure.

File: model/jvm/src/main/resources/edit-metadata.txt [hunk 2]
```diff
@@ -103,7 +103,7 @@ LAR,Quality,Q037,"If lien status = 2, then loan amount should be ≤ $250 ($250
...
+LAR,Quality,Q068,"If action taken type = 1-5, 7 or 8, and applicant ethnicity, race and sex = 4,7,4 respectively, and co-applicant ethnicity, race and sex = 4,7,4, respectively, and there is no co-applicant, then co-applicant ethnicity, race, and sex should = 5,8,5, respectively.","The loan is reported as not purchased by your institution (action taken type = 1-5, 7 or 8), with no applicant (applicant ethnicity = 4, applicant race = 7, applicant sex = 4), and co-applicant as not applicable (co-applicant ethnicity = 4, co-applicant race = 7, co-applicant sex = 4). If there is no co-applicant on the loan, the co-applicant data fields should instead indicate that there is no co-applicant (co-applicant ethnicity = 5, co-applicant race = 8, co-applicant sex = 5). Please verify and update your file if needed. ",Type of Action Taken; Applicant Ethnicity; Applicant Race: 1; Applicant Sex; Co-applicant Ethnicity; Co-applicant Race: 1; Co-applicant Sex
```
Comment (schbetsy) [inline model/jvm/src/main/resources/edit-metadata.txt]: What changed here?

Comment (nickgrippin) [inline model/jvm/src/main/resources/edit-metadata.txt]: I honestly don't know, I tried comparing the lines and there's no difference

File: api/src/main/scala/hmda/api/http/ValidationErrorConverter.scala (hunk 1 above)
Comment (nickgrippin) [inline api/src/main/scala/hmda/api/http/ValidationErrorConverter.scala]: Done in 5bc621096da0b8619c1a47b36679a46b91f8691d

File: model/jvm/src/main/resources/edit-metadata.txt (hunk 2 above)
Comment (nickgrippin) [inline model/jvm/src/main/resources/edit-metadata.txt]: Ah, on second look, there's an extra space that was trimmed off the end of each of these lines.  Happened automatically on file save in IntelliJ.

**Pull request merged**